core/controllers/reader_test.py

```diff
@@ -205,6 +248,103 @@ def test_huge_answers(self):
             'a' * 1000500, 'Sorry, nope, we didn\'t get it', '')
 
 
+class ReaderClassifyTests(test_utils.GenericTestBase):
+    """Test reader.classify using the sample explorations.
+
+    Since the end to end tests cover correct classification,
+    ReaderClassifyTests is only checking which of the hard/soft/classifier
+    rules triggered on an input.
+    """
+
+    def setUp(self):
+        super(ReaderClassifyTests, self).setUp()
+        self.init_classify_inputs('16')
+
+    def init_classify_inputs(self, exploration_id):
+        exp_services.delete_demo(exploration_id)
+        exp_services.load_demo(exploration_id)
+
+        self.EXP_ID = exploration_id
+        self.EXP_STATE = (
+            exp_services.get_exploration_by_id(exploration_id).states['Home'])
```

Comment:
You don't need to call ``exp_services.get_exploration_by_id()`` since ``exp_services.load_demo()`` returns the exploration. Feel free to save the result in a local exploration variable and use that to extract the Home state.